meteor/client/lib/ShowStyleVariantImportButton.tsx

```diff
@@ -0,0 +1,168 @@
+import { ShowStyleVariant } from '../../lib/collections/ShowStyleVariants'
+import { doModalDialog } from './ModalDialog'
+import { MeteorCall } from '../../lib/api/methods'
+import { logger } from '../../lib/logging'
+import React from 'react'
+import { t } from 'i18next'
+import { NoticeLevel, Notification, NotificationCenter } from './notifications/notifications'
+import { UploadButton } from './uploadButton'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { faUpload } from '@fortawesome/free-solid-svg-icons'
+import { withTranslation } from 'react-i18next'
+
+interface IShowStyleVariantImportProps {
+	showStyleVariants: ShowStyleVariant[]
+}
+
+interface IShowStyleVariantImportState {
+	timestampedFileKey: number
+}
+
+export const ShowStyleVariantImportButton = withTranslation()(
+	class ShowStyleVariantImportButton extends React.Component<
+		IShowStyleVariantImportProps,
+		IShowStyleVariantImportState
+	> {
+		constructor(props: IShowStyleVariantImportProps) {
+			super(props)
+
+			this.state = {
+				timestampedFileKey: Date.now(),
+			}
+		}
+
+		private getDuplicatedShowStyleVariantAmount(showStyleVariant: ShowStyleVariant): number {
```

Comment:
Will this ever return higher than 1?
If no, why not then make it return a boolean and call it something like `alreadyContainsShowStyleVariant`?

Reply:
Nvm, I see what you use this for :)

Reply:
Sidenote: In order to make a nice and a readable flow through the class I highly recommend that you place your methods in the order they are used. This makes it so you don't have to scroll back and forth (as much) when reading a class.

It looks something like this:
```
public functionA() {
  functionB()
  functionC()
}

private functionB() {
  functionD()
}

private functionD() {
  ...
}

private functionC() {
  ...
}
```

Reply:
I have the changed the order to more accurately depict the call hierarchy!